static/app/components/group/tagFacets/index.tsx

```diff
@@ -103,14 +113,44 @@ export default function TagFacets({
     }
 
     const keyed = keyBy(data, 'key');
-    const formatted = tagFormatter?.(keyed) ?? keyed;
+    // Statistical detector issues need to use a Discover query
+    // which means we need to massage the values to fit the component API
+    if (isStatisticalDetector) {
+      Object.keys(keyed).forEach(tagKey => {
```

Comment:
can we put this into a util function for clarity

Reply:
also making a comment that we're mutating `keyed` when invoking that util function

Reply:
I've put it into a util function and instead of mutating `keyed`, I return a new object now